nautobot_device_lifecycle_mgmt/management/commands/generate_app_test_data.py

```diff
@@ -0,0 +1,137 @@
+"""Generate test data for the Golden Config app."""
+
+import random
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+from django.db import DEFAULT_DB_ALIAS
+from nautobot.core.factory import get_random_instances
+from nautobot.dcim.models import Device, DeviceType, InventoryItem, Platform
+
+from nautobot_device_lifecycle_mgmt.choices import CVESeverityChoices
+from nautobot_device_lifecycle_mgmt.models import (
+    CVELCM,
+    ContactLCM,
+    ContractLCM,
+    DeviceSoftwareValidationResult,
+    HardwareLCM,
+    InventoryItemSoftwareValidationResult,
+    ProviderLCM,
+    SoftwareImageLCM,
+    SoftwareLCM,
+    ValidatedSoftwareLCM,
+    VulnerabilityLCM,
+)
+
+
+class Command(BaseCommand):
+    """Populate the database with various data as a baseline for testing (automated or manual)."""
+
+    help = __doc__
+
+    def add_arguments(self, parser):  # noqa: D102
+        parser.add_argument(
+            "--flush",
+            action="store_true",
+            help="Flush any existing data from the database before generating new data.",
+        )
+        parser.add_argument(
+            "--database",
+            default=DEFAULT_DB_ALIAS,
+            help='The database to generate the test data in. Defaults to the "default" database.',
+        )
+
+    def _generate_static_data(self):
+        devices = get_random_instances(Device, minimum=2, maximum=4)
```

Comment:
Since this requires the factory for devices which wasn't put in till 2.2.0, I believe this should be in a try/except to catch and return a user friendly error if there are no devices. Or, you can try to import the DeviceFactory.

Reply:
I updated the docs instead